sycl/test/atomic_ref/compare_exchange.cpp

```diff
@@ -60,12 +62,33 @@ int main() {
   }
 
   constexpr int N = 32;
+  // CHECK-LLVM: declare dso_local spir_func i32
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicCompareExchange
+  // CHECK-LLVM-SAME: (i32 addrspace(1)*, i32, i32, i32, i32, i32)
   compare_exchange_test<int>(q, N);
+  // CHECK-LLVM: declare dso_local spir_func i32
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicCompareExchange
+  // CHECK-LLVM-SAME: (i32 addrspace(1)*, i32, i32, i32, i32, i32)
   compare_exchange_test<unsigned int>(q, N);
+  // CHECK-LLVM: declare dso_local spir_func i[[long:(32)|(64)]]
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicCompareExchange
+  // CHECK-LLVM-SAME: (i[[long]] addrspace(1)*, i32, i32, i32,
+  // CHECK-LLVM-SAME: i[[long]], i[[long]])
   compare_exchange_test<long>(q, N);
+  // CHECK-LLVM: declare dso_local spir_func i[[long]]
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicCompareExchange
+  // CHECK-LLVM-SAME: (i[[long]] addrspace(1)*, i32, i32, i32,
+  // CHECK-LLVM-SAME: i[[long]], i[[long]])
   compare_exchange_test<unsigned long>(q, N);
+  // CHECK-LLVM: declare dso_local spir_func i64
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicCompareExchange
+  // CHECK-LLVM-SAME: (i64 addrspace(1)*, i32, i32, i32, i64, i64)
   compare_exchange_test<long long>(q, N);
+  // CHECK-LLVM: declare dso_local spir_func i64
+  // CHECK-LLVM-SAME: @_Z{{[0-9]+}}__spirv_AtomicCompareExchange
+  // CHECK-LLVM-SAME: (i64 addrspace(1)*, i32, i32, i32, i64, i64)
   compare_exchange_test<unsigned long long>(q, N);
+  // The remaining functions use the already-declared ones on the IR level
```

Comment:
Do they? It was integers and now we have floats and doubles...

Reply:
The reason for this is that float/pointer-typed functions are [emulated](https://github.com/intel/llvm/blob/7a8e02a1e26ee6160c112581d8851272e35e4dc9/sycl/include/CL/sycl/ONEAPI/atomic_ref.hpp#L454) instead of being mapped onto particular SPIR-V operands. I do not know the details of why integer `compare_exchange` external calls are made inside this emulation, but #2765 exemplifies what happens if the emulation is changed by an external call to a unique float-typed function.